### DISCUSSION TITLE
Enthalpy in eNRTL property package

### DISCUSSION THREAD

**Discussion by am000-simreka** [IDAES/idaes-pse#1491] [Q&A]
Dear IDAES team,
I am using the example enrtl_H2O_NaCl_KCl.py (in ideas-pse/models/properties/modular_properties/examples) to build a property package for a process simulation of a hydrogen production case. The full list of components are: H2O liquid, KOH, H2 gas/liquid O2 gas/liquid N2 gas, K+ and OH-. For now, I am just trying to add KOH, H2O (solvent) and K+ and OH-; I have fetched the tau values from [1]. Now I receive the error:

_Generic Property Package instance fs.F101.control_volume.properties_in[0.0] called for enth_mol_liq_comp, but was not provided with a method for this property._

I have recently found an expression in [2] that describes a full enthalpy model using the heat capacity of the ionic species. However, I am not certain if this is the best practice for eNRTL. Can you provide further insight into how to set up this configuration file?

References
[1] Local Composition Model for Excess Gibbs Energy of Electrolyte Systems, Pt 1.     Chen, C.-C., Britt, H.I., Boston, J.F., Evans, L.B.,AIChE Journal, 1982, Vol. 28(4), pgs. 588-596.
[2] A modified electrolyte non-random two-liquid model with analytical expression for excess enthalpy: Application to the MEA-H2O-CO2 system. Paul Akula, Andrew Lee,  John Eslick,  Debangsu Bhattacharyya,  David C. Miller. ttps://doi.org/10.1002/aic.17935

**Comment by andrewlee94** [comment]
@am000-simreka The root cause of the error message is that you need to provide a method which describes the pure component enthalpies for all species in your system. It theory at least, this is entirely separate from the departure functions related to the non-ideal mixing effects and can be modeled in any way you wish. However, these equations are also very case specific (and are often essentially empirical correlations), and are difficult to define for ionic species (as they never exist in pure form).

The paper you referred to may be a useful starting point, but unfortunately it is too long since I read it to be able to say much about it. Note however the the MEA-H2O-CO2 system is rather different from a simpler binary salt system.
